### PR TITLE
feat: disable create account with proxy [AR-2786]

### PR DESCRIPTION
**Pull request by MohammedMokresh**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2786" title="AR-2786" target="_blank">TaskAR-2786</a>  Disable create account while using proxy
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

If the user have the proxy enabled using the deeplink we are not supporting create acount or team or login with SSO for now

### Solutions

Show dialogs for the user when trying to create account to tell that it's not possible

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Use a deeplink where a proxy supported, try to create account or login with SSO, you should not be about to and see alert message

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1213/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5136056)(5136056)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5136056)(5136056)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1213/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5136553)(5136553)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5136553)(5136553)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/authentication/login/LoginScreen.kt [hunk 1]
```diff
@@ -94,7 +101,21 @@ private fun LoginContent(
                 WireTabRow(
                     tabs = LoginTabItem.values().toList(),
                     selectedTabIndex = pagerState.calculateCurrentTab(),
-                    onTabChange = { scope.launch { pagerState.animateScrollToPage(it) } },
+                    onTabChange = {
+
+                        if (viewModel.serverConfig.apiProxy != null) {
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/authentication/login/LoginScreen.kt]: how about moving this flag to a viewmodel and check for a null there and expose it through a state or something ?

File: app/src/main/kotlin/com/wire/android/ui/authentication/welcome/WelcomeScreen.kt [hunk 2]
```diff
@@ -116,15 +123,36 @@ private fun WelcomeContent(viewModel: WelcomeViewModel) {
...
                 LoginButton(viewModel::goToLogin)
+                FeatureDisabledWithProxyDialogContent(dialogState = enterpriseDisabledWithProxyDialogState,
+                    onActionButtonClicked = {
+                        CustomTabsHelper.launchUrl(context, viewModel.state.teams)
+                    })
+                FeatureDisabledWithProxyDialogContent(dialogState = createPersonalAccountDisabledWithProxyDialogState)
 
                 CreateEnterpriseAccountButton {
-                    viewModel.goToCreateEnterpriseAccount()
+                    if (viewModel.state.apiProxy != null) {
```
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/authentication/welcome/WelcomeScreen.kt]: same here how about moving this to a viewmodel ?

**Review by trOnk12 — APPROVED** [review]
minor comments to move some stuff to a viewmodel instead of doing it in the UI

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1213/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5143863)(5143863)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5143863)(5143863)-dev-release.apk)

**Review by MohamadJaara — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1213/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(5145397)(5145397)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(5145397)(5145397)-dev-release.apk)

**Pull request merged**